server/src/main/scala/controllers/application/WithImports.scala

```diff
@@ -31,17 +34,34 @@ trait WithImports {
       log.info(s"Received a request to import red list counts")
       request.body.asJson match {
         case Some(content) =>
-          log.info(s"Received red list pax data")
+          log.info(s"Received red list pax data $content")
           Try(content.toString.parseJson.convertTo[RedListCounts])
-            .map(redListCounts => ctrl.flightsActor
-              .ask(redListCounts)
-              .map(_ => Accepted(toJson(ApiResponseBody(s"${redListCounts.counts.size} red list records imported"))))
-            ).getOrElse(Future.successful(BadRequest("Failed to parse json")))
+            .map { redListCounts =>
+              getRedListCount(redListCounts)
+                .onComplete {
+                  case Success(updatedRedListCounts) => ctrl.flightsActor
+                    .ask(RedListCounts(updatedRedListCounts))
+                  case Failure(exception) =>
+                    log.error(s"Error $exception ")
+                }
+              Future.successful(Accepted(toJson(ApiResponseBody(s"${redListCounts.passengers.size} red list records imported"))))
+            }.getOrElse(Future.successful(BadRequest("Failed to parse json")))
         case None => Future.successful(BadRequest("No content"))
       }
     }
   }
 
+  def getRedListCount(redListCounts: RedListCounts): Future[Iterable[RedListPassengers]] = Future.sequence {
+    redListCounts.passengers
+      .map { redListPassenger =>
+        val actor: ActorRef = system.actorOf(NeboArrivalActor.props(redListPassenger, now))
+        val stateF: Future[NeboArrivals] = actor.ask(redListPassenger).mapTo[NeboArrivals]
```

Comment:
We need to remember to kill the actor after we've sent the update and got the response